RELEASE/scripts/autoscend/auto_providers.ash

```diff
@@ -1212,3 +1212,336 @@ boolean provideMoxie(int amt, boolean doEquips)
+	{
+		if(!speculative)
+			auto_sourceTerminalEnhance("meat");
+		handleEffect($effect[meat.enh]); //60% meat
+		if(pass())
+			return result();
+	}
+	if(item_amount($item[body spradium]) > 0 && !in_tcrs() && have_effect($effect[Boxing Day Glow]) == 0)
+	{
+		autoChew(1, $item[body spradium]); //50% meat, 5 fam weight
+		if(pass())
+			return result();
+	}
+
+	// craft equipment, even limited use, here
+	if(doEverything)
+	{
+		handleBjornify($familiar[Hobo Monkey]); //25% meat, hot damage, delevels
+		//craft IOTM derivative that gives high item bonus
+		if((equipped_item($slot[off-hand]) != $item[Half a Purse]) && !possessEquipment($item[Half a Purse]) && (item_amount($item[Lump of Brituminous Coal]) > 0))
+		{
+			//+X% meat based on smithness (10% if only half a purse is equipped)
+			auto_buyUpTo(1, $item[Loose Purse Strings]);
+			autoCraft("smith", 1, $item[Lump of Brituminous Coal], $item[Loose purse strings]);
+		}
+		string max = "500meat " + (amt + 100) + "max";
+		if(speculative)
+		{
+			simMaximizeWith(loc, max);
+		}
+		else
+		{
+			addToMaximize(max);
+			simMaximize(loc);
+		}
+		delta = simValue("Meat Drop") - numeric_modifier("Meat Drop");
+		auto_log_debug("With existing and crafted gear we can get to " + result());
+		if(pass())
+			return result();
+	}
+	// Use limited resources like Inhaler
+	if(doEverything)
+	{
+		if(tryEffects($effects[
+		shadow waters, //200% meat, 100% item, 100% init, -10% combat
+		Sinuses For Miles, //200% meat
+		Car-Charged, //100% meat, 100% item, 5-10MP, 50% init, 50% spell dmg, +3 stats per fight
+		Incredibly Well Lit //100% meat, 50% item
+		]))
+			if(pass())
+				return result();
+		if(zataraAvailable() && (0 == have_effect($effect[Meet the Meat])) & auto_is_valid($effect[Meet the Meat]))
+		{
+			if(!speculative)
+			{
+				zataraSeaside("meat");
+			}
+			handleEffect($effect[Meet the meat]); //100% meat, 50% gear drops
+			if(pass())
+				return result();			
+		}
+		if((get_property("sidequestArenaCompleted") == "fratboy") && !get_property("concertVisited").to_boolean() && (have_effect($effect[Winklered]) == 0))
+		{
+			if(is_professor())
+			{
+				//Need to manually equip because professor
+				if(!have_equipped($item[beer helmet])) equip($item[beer helmet]);
+				if(!have_equipped($item[distressed denim pants])) equip($item[distressed denim pants]);
+				if(!have_equipped($item[bejeweled pledge pin])) equip($item[bejeweled pledge pin]);
+			}
+			else
+			{
+				outfit("Frat Warrior Fatigues");
+			}
+			if(!speculative)
+			{
+				cli_execute("concert 2"); //40% meat
+			}
+			handleEffect($effect[Winklered]); //40% meat
+			if(pass())
+				return result();
+		}
+		if(pass())
+			return result();
+		if(auto_monkeyPawWishesLeft() + auto_wishesAvailable() > 0)
+		{
+			boolean success = true;
+			int specwishes = 0;
+			while(auto_monkeyPawWishesLeft() + auto_wishesAvailable() > 0 && success)

```

Comment:
is the while doing anything since we have a foreach?

Reply:
No it's not